contracts/option_pricing.cairo

```diff
@@ -166,11 +142,9 @@ func d1_d2{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
 
     alloc_locals;
 
-    let TWO = Math64x61.fromFelt(2);
-
     let sqrt_time_till_maturity_annualized = Math64x61.sqrt(time_till_maturity_annualized);
     let sigma_squared = Math64x61.mul(sigma, sigma);
-    let sigma_squared_half = Math64x61.div(sigma_squared, TWO);
+    let sigma_squared_half = Math64x61.mul(sigma_squared, HALF);
```

Comment:
Curious, what’s the rationale behind this?

Reply:
saves 42 steps, 2 memmory holes and 4 range_check_builtin

Reply:
*on two passes (the black scholes is called twice in the test that I used to measure this)... so half of this for one pass